### PR TITLE
Add more options for setting up kind clusters

### PR DESCRIPTION
**Pull request by jacob-delgado**
**Please provide a description of this PR:**

Improve setupkind.sh

### CODE REVIEW

File: samples/kind-lb/README.md [hunk 1]
```diff
@@ -17,11 +17,16 @@ Usage:
...
 
 Where:
-  -n|--cluster-name - name of the k8s cluster to be created
-  -r|--k8s-release  - the release of the k8s to setup, latest available if not given
-  -s|--ip-space     - the 2rd to the last part for public ip addresses, 255 if not given, valid range: 0-255
-  -i|--ip-family    - ip family to be supported, default is ipv4 only. Value should be ipv4, ipv6, or dual
-  -h|--help         - print the usage of this script
+  -n|--cluster-name  - name of the k8s cluster to be created
+  -r|--k8s-release   - the release of the k8s to setup, latest available if not given
+  -s|--ip-space      - the 2rd to the last part for public ip addresses, 255 if not given, valid range: 0-255.
```
Comment (dhawton) [inline samples/kind-lb/README.md:22]:
```suggestion
  -s|--ip-space      - the 2nd to the last part for public ip addresses, 255 if not given, valid range: 0-255.
```

File: samples/kind-lb/README.md [hunk 2]
```diff
@@ -17,11 +17,16 @@ Usage:
...
+  -m|--mode          - setup the required number of nodes per deployment model. Values are sidecar (1 node) or ambient (minimum of 2)
+  -w|--worker-nodes  - the number of worker nodes to create. Default is 1
+  --pod-subnet       - the pod subnet to specify. Default is 10.244.0.0/16 for IPv4 and fd00:10:244::/56 for IPv6
+  --service-subnet   - the service subnet to specify. Default is 10.96.0.0/16 for IPv4 and fd00:10:96::/112 for IPv6
+  -i|--ip-family     - ip family to be supported, default is ipv4 only. Value should be ipv4, ipv6, or dual (ipv4-preferred by default)
```
Comment (dhawton) [inline samples/kind-lb/README.md:27]: Default doesn't match the list of values.

File: samples/kind-lb/setupkind.sh [hunk 3]
```diff
@@ -29,8 +29,13 @@ function printHelp() {
   echo "Where:"
   echo "    -n|--cluster-name  - name of the k8s cluster to be created"
   echo "    -r|--k8s-release   - the release of the k8s to setup, latest available if not given"
-  echo "    -s|--ip-space      - the 2rd to the last part for public ip addresses, 255 if not given, valid range: 0-255"
-  echo "    -i|--ip-family     - ip family to be supported, default is ipv4 only. Value should be ipv4, ipv6, or dual"
+  echo "    -s|--ip-space      - the 2rd to the last part for public ip addresses, 255 if not given, valid range: 0-255."
```
Comment (dhawton) [inline samples/kind-lb/setupkind.sh:32]:
```suggestion
  echo "    -s|--ip-space      - the 2nd to the last part for public ip addresses, 255 if not given, valid range: 0-255."
```

File: samples/kind-lb/setupkind.sh [hunk 4]
```diff
@@ -29,8 +29,13 @@ function printHelp() {
...
+  echo "    -m|--mode          - setup the required number of nodes per deployment model. Values are sidecar (1 node) or ambient (minimum of 2)"
+  echo "    -w|--worker-nodes  - the number of worker nodes to create. Default is 1"
+  echo "    --pod-subnet       - the pod subnet to specify. Default is 10.244.0.0/16 for IPv4 and fd00:10:244::/56 for IPv6"
+  echo "    --service-subnet   - the service subnet to specify. Default is 10.96.0.0/16 for IPv4 and fd00:10:96::/112 for IPv6"
+  echo "    -i|--ip-family     - ip family to be supported, default is ipv4 only. Value should be ipv4, ipv6, or dual (ipv4-preferred by default)"
```
Comment (dhawton) [inline samples/kind-lb/setupkind.sh:37]: Same here about default and list of values

File: samples/kind-lb/setupkind.sh [hunk 5]
```diff
@@ -106,56 +124,63 @@ for family in "${validIPFamilies[@]}"; do
   fi
 done
 
+
```
Comment (dhawton) [inline samples/kind-lb/setupkind.sh:127]:
```suggestion
```

Double blank line

File: samples/kind-lb/setupkind.sh [hunk 6]
```diff
@@ -29,8 +29,13 @@ function printHelp() {
...
   echo "    -n|--cluster-name  - name of the k8s cluster to be created"
   echo "    -r|--k8s-release   - the release of the k8s to setup, latest available if not given"
-  echo "    -s|--ip-space      - the 2rd to the last part for public ip addresses, 255 if not given, valid range: 0-255"
-  echo "    -i|--ip-family     - ip family to be supported, default is ipv4 only. Value should be ipv4, ipv6, or dual"
+  echo "    -s|--ip-space      - the 2nd to the last part for public ip addresses, 255 if not given, valid range: 0-255."
+  echo "    -m|--mode          - setup the required number of nodes per deployment model. Values are sidecar (1 node) or ambient (minimum of 2)"
+  echo "    -w|--worker-nodes  - the number of worker nodes to create. Default is 1"
+  echo "    --pod-subnet       - the pod subnet to specify. Default is 10.244.0.0/16 for IPv4 and fd00:10:244::/56 for IPv6"
+  echo "    --service-subnet   - the service subnet to specify. Default is 10.96.0.0/16 for IPv4 and fd00:10:96::/112 for IPv6"
+  echo "    -i|--ip-family     - ip family to be supported, default is ipv4 only. Value should be ipv4, ipv6, or dual (ipv4-preferred by default)"
```
Comment (jacob-delgado) [inline samples/kind-lb/setupkind.sh:37]:
```suggestion
  echo "    -i|--ip-family     - ip family to be supported, default is ipv4 only. Value should be ipv4, ipv6, or dual"
```

**Review by ericvn — APPROVED** [review]

**Pull request merged**